ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionLogUpdatesConversions.scala

```diff
@@ -0,0 +1,409 @@
+            childEventIds = exercisedEvent.children,
+            exerciseResult = maybeExerciseResult,
+          )
+        )
+      )
+    }
+
+    private def createdToTransactionTreeEvent(
+        requestingParties: Set[Party],
+        verbose: Boolean,
+        lfValueTranslation: LfValueTranslation,
+        createdEvent: CreatedEvent,
+    )(implicit
+        loggingContext: LoggingContext,
+        executionContext: ExecutionContext,
+    ) = {
+      val eventualContractKey = createdEvent.contractKey
+        .map(
+          lfValueTranslation
+            .toApiValue(
+              _,
+              verbose,
+              "create key",
+              value =>
+                lfValueTranslation.enricher
+                  .enrichContractKey(createdEvent.templateId, value.value),
+            )
+            .map(Some(_))
+        )
+        .getOrElse(Future.successful(None))
+
+      val eventualCreateArguments = lfValueTranslation.toApiRecord(
+        createdEvent.createArgument,
+        verbose,
+        "create argument",
+        value =>
+          lfValueTranslation.enricher
+            .enrichContract(createdEvent.templateId, value.value),
+      )
+
+      for {
+        maybeContractKey <- eventualContractKey
+        createArguments <- eventualCreateArguments
+      } yield TreeEvent(
+        TreeEvent.Kind.Created(
+          com.daml.ledger.api.v1.event.CreatedEvent(
+            eventId = createdEvent.eventId.toLedgerString,
+            contractId = createdEvent.contractId.coid,
+            templateId = Some(LfEngineToApi.toApiIdentifier(createdEvent.templateId)),
+            contractKey = maybeContractKey,
+            createArguments = Some(createArguments),
+            witnessParties = createdEvent.treeEventWitnesses
+              .intersect(requestingParties.map(_.toString))
+              .toSeq,
+            signatories = createdEvent.createSignatories.toSeq,
+            observers = createdEvent.createObservers.toSeq,
+            agreementText = createdEvent.createAgreementText.orElse(Some("")),
+          )
+        )
+      )
+    }
+
+    private val TransactionTreePredicate: Set[Party] => TransactionLogUpdate.Event => Boolean =
+      requestingParties => {
+        case createdEvent: CreatedEvent =>
+          createdEvent.treeEventWitnesses
+            .intersect(requestingParties.map(_.toString))
+            .nonEmpty
+        case exercised: ExercisedEvent =>
+          exercised.treeEventWitnesses
+            .intersect(requestingParties.map(_.toString))
+            .nonEmpty
+      }
+  }
+
+  private def instantToTimestamp(t: Instant): Timestamp =
+    Timestamp(seconds = t.getEpochSecond, nanos = t.getNano)
+
+  private def getCommandId(
+      events: Vector[TransactionLogUpdate.Event],
+      commandId: String,
+      requestingParties: Set[Party],
+  ) =
+    events
+      .find(_.commandId != "")
+      // TODO // TDT Double check condition for command id setting
```

Comment:
The logic for setting the `commandId` is based on two assumptions:
* `submitters` is never empty for a flat transaction event. For my knowledge though: can `submitters` ever be empty?
* The `requesting` parties should contain at least one of the `submitters`. This condition mimics the logic as implemented in the SQL predicate [here](https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEventsRangeQueries.scala#L216)

Reply:
CC @meiersi-da @mziolekda ?